### ISSUE TITLE
DynamoDBContext.FromDocument Exception for string without milliseconds (decimal seconds)

### ISSUE THREAD

**Issue by mlhpdx** [aws/aws-sdk-net#2020]
### Describe the bug

DynamoDBContext.FromDocument will throw an exception converting a valid ISO date/time string (for example "2022-05-05T11:59:11Z") to a DateTime object. The code works correctly only when the decimal seconds are included, for example the equivalent string "2022-05-05T11:59:11.000Z".

This behavior is problematic in using DynamoDBContext with data sources from other AWS services that may remove trailing zeros. See https://repost.aws/questions/QU6ga7E0zPSRKIpqx1OEIFOA/incompatibility-between-state-entered-time-and-the-dynamo-db-net-sdk.  

### Expected Behavior

The DateTime will be correctly deserialized from a string with or without decimal seconds, and model object returned without an exception.

### Current Behavior

Currently an exception is thrown when attempting to deserialize the string to a DateTime:

```Exception has occurred: CLR/System.InvalidOperationException
An exception of type 'System.InvalidOperationException' occurred in AWSSDK.DynamoDBv2.dll but was not handled in user code: 'Unable to convert [2022-05-05T11:56:11Z] of type Amazon.DynamoDBv2.DocumentModel.Primitive to System.DateTime'
   at Amazon.DynamoDBv2.Converter.FromEntry(DynamoDBEntry entry, Type targetType)
```

### Reproduction Steps

In the form of a unit test, the problem is reproducible (test will fail with an exception) with:

```
        public class TestObject
        {
            public DateTime DateFromString { get; set; }
        }

        [Fact]
        public void DateTimeDeserializationWithDdbContext()
        {
            Environment.SetEnvironmentVariable("AWS_REGION", "us-west-2");
            Environment.SetEnvironmentVariable("AWS_PROFILE", "not-a-real-profile");

            var json = JsonConvert.SerializeObject(new {
                DateFromString = "2022-05-05T11:56:11Z"
            });
            var doc = Document.FromJson(json);
                        
            var model = new DynamoDBContext(new AmazonDynamoDBClient()).FromDocument<TestObject>(doc);

            Assert.NotNull(model);
        }

```

### Possible Solution

The use of `DateTime.ParseExact` seems overly restrictive, and creates a situation of incompatibility with other AWS services that store data/time values.  Switching to simple `DateTime.Parse` would seem a safe and pragmatic choice. Any code that would break with this change would seem to be dependent on a bug (exceptions on deserializing valid dates).

Change:
```
    public object FromEntry(DynamoDBEntry entry)
    {
        return DateTime.ParseExact(entry.AsString(), "o", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind);
    }
```
To:
```
    public object FromEntry(DynamoDBEntry entry)
    {
        return DateTime.Parse(entry.AsString(), CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind);
    }
```

### Additional Information/Context

As more and more "direct" integrations between services are employed by AWS users, it is important that interoperability be considered a very high priority. This bug is one that make such interoperation more complicated and error prone.  Even more so in the example of the behavior of Step Functions, which is unpredictable but correct (complied with ISO date/time formatting) and only causes this problem when a time is exactly "on a second") -- i.e. 1 time in 1000 executions, give or take.

### AWS .NET SDK and/or Package version used

AWSSDK.DynamoDBv2 3.7.0.70

### Targeted .NET Platform

.Net 6

### Operating System and version

Windows, Ubuntu, Amazon Linux 2

**Comment by ashishdhingra** [comment]
Reproducible.

**Comment by peterrsongg** [comment]
@mlhpdx This fix was released in dynamodb version 3.7.101.48 on 2023-02-18. Closing this issue as it has been resolved. Feel free to open it if there is still an issue

**Comment by lee-11** [comment]
> @mlhpdx This fix was released in dynamodb version 3.7.101.48 on 2023-02-18.

Fantastic, and thank you!
